daily_milestone.go

```diff
@@ -1,29 +1,221 @@
+	return list, nil
+}
+
+// CreateMilestoneData is used to check the due date using the time package of python
+func createMilestoneData(advance int) []string {
+	today := time.Now().Local()
+	list := []string{}
+	for i := 0; i < advance; i++ {
+		date := today.AddDate(0, 0, i)                                                                      // To Format to ISOFormat and it converts to string so can be used in list directly
+		datelist := []string{"Title: ", date.Format("2006-01-02"), "  due_date", date.Format("2006-01-02")} // Was unable to get a map in a list so made this
+		y := strings.Join(datelist, ",")
+		list = append(list, y)
+	}
+
+	fmt.Println(list)
+	return list
+}
+
+func createMilestones(baseURL string, token string, projectID string, milestones []string) (string, error) {
+	strurl := []string{baseURL, projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	for _, m := range milestones {
+		mbyte := bytes.NewReader([]byte(m))
+		req, err := http.NewRequest("POST", url, mbyte)
+		if err != nil {
+			return "", err
+		}
+		req.Header.Add("PRIVATE-TOKEN", token)
+		resp, err := client.Do(req)
+		if err != nil {
+			return "", err
+		}
+		defer resp.Body.Close()
+	}
+	return ("Milestones Created: " + strings.Join(milestones, "")), nil
+}
 
 func main() {
-	var apiKey, apiBase, namespace, project string
-	var advance int
-	flag.StringVar(&apiKey, "token", " ", "Gitlab api key/token.")
-	flag.StringVar(&apiBase, "base_url", " ", "Gitlab api base url")
-	flag.StringVar(&namespace, "namespace", " ", "Namespace to use in Gitlab")
-	flag.StringVar(&project, "projectname", " ", "Project to use in Gitlab")
-	flag.IntVar(&advance, "Advance", 30, "Define timeframe to generate milestones in advance.")
-	flag.Parse()
+	// Declaring variables for flags
+	var Token, baseURL, Namespace, Project string
+	var Advance int
+	// Command Line Parsing Starts
+	flag.StringVar(&Token, "Token", "bVYFTJaYtgAZesSofKbq", "Gitlab api key/token.")
+	flag.StringVar(&baseURL, "baseURL", "dev.seetheprogress.eu", "Gitlab api base url")
+	flag.StringVar(&Namespace, "Namespace", "okkur", "Namespace to use in Gitlab")
+	flag.StringVar(&Project, "ProjectName", "dailymile_test", "Project to use in Gitlab")
+	flag.IntVar(&Advance, "Advance", 30, "Define timeframe to generate milestones in advance.")
+	flag.Parse() //Command Line Parsing Ends
+
+	// Initializing logger
+	LoggerSetup(os.Stdout)
+	// Calling getProjectID
+	baseurl := "https://" + baseURL + "/api/v4" + "/projects/"
+	projectID, err := getProjectID(baseurl, Token, Project, Namespace)
+	if err != nil {
+		logger.Println(err)
+	}
+	oldMilestone, err := getMilestones(baseurl, Token, projectID)
+	if err != nil {
+		logger.Println(err)
+	}
+	newMilestone := createMilestoneData(Advance)
+
+	for index, y := range newMilestone {
+		for _, z := range oldMilestone {
+			if z == y {
+				newMilestone = append(newMilestone[:index], newMilestone[(index+1):]...)
+			}
+		}
 
+	}
```

Comment:
Whitespace here too

Reply:
Yeah I don't know why this is happening. I don't get it on atom. Any idea?

Reply:
Might be because of windows vs Linux line return ? Try to edit the file in vim/nano